pkg/operator/bootstrapteardown/teardown.go

```diff
@@ -4,61 +4,19 @@ import (
 	"context"
 	"time"
 
-	configv1 "github.com/openshift/api/config/v1"
 	operatorv1 "github.com/openshift/api/operator/v1"
-	configclient "github.com/openshift/client-go/config/clientset/versioned"
 	etcdv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
 	"github.com/openshift/cluster-etcd-operator/pkg/operator/clustermembercontroller"
-	cov1helpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/apimachinery/pkg/watch"
+
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/cache"
-	clientwatch "k8s.io/client-go/tools/watch"
 	"k8s.io/klog"
 )
 
 func TearDownBootstrap(config *rest.Config,
 	clusterMemberShipController *clustermembercontroller.ClusterMemberController, etcdClient etcdv1.EtcdInterface) error {
-	failing := configv1.ClusterStatusConditionType("Failing")
-	var lastError string
-
-	cc := configclient.NewForConfigOrDie(config)
-	_, err := clientwatch.UntilWithSync(
-		context.Background(),
-		cache.NewListWatchFromClient(cc.ConfigV1().RESTClient(), "clusterversions", "", fields.OneTermEqualSelector("metadata.name", "version")),
-		&configv1.ClusterVersion{},
-		nil,
-		func(event watch.Event) (bool, error) {
-			switch event.Type {
-			case watch.Added, watch.Modified:
-				cv, ok := event.Object.(*configv1.ClusterVersion)
-				if !ok {
-					klog.Warningf("Expected a ClusterVersion object but got a %q object instead", event.Object.GetObjectKind().GroupVersionKind())
-					return false, nil
-				}
-				if cov1helpers.IsStatusConditionTrue(cv.Status.Conditions, configv1.OperatorAvailable) {
-					return true, nil
-				}
-				if cov1helpers.IsStatusConditionTrue(cv.Status.Conditions, failing) {
-					lastError = cov1helpers.FindStatusCondition(cv.Status.Conditions, failing).Message
-				} else if cov1helpers.IsStatusConditionTrue(cv.Status.Conditions, configv1.OperatorProgressing) {
-					lastError = cov1helpers.FindStatusCondition(cv.Status.Conditions, configv1.OperatorProgressing).Message
-				}
-				klog.Errorf("Still waiting for the cluster to initialize: %s", lastError)
-				return false, nil
-			}
-			klog.Infof("Still waiting for the cluster to initialize...")
-			return false, nil
-		},
-	)
-
-	if err != nil {
-		klog.Errorf("error in watching clusterversions: %#v", err)
-	}
-
+	err := WaitForEtcdBootstrap(context.TODO(), config)
```

Comment:
why do we disregard these errors?

Reply:
+1, good catch

No reason to ignore the error should catch it with an if block

Reply:
done